pkgs/default.nix

```diff
@@ -104,16 +120,20 @@ let
           "SC2143"
           "SC2148"
         ];
+        prelude = ''
+          #!${pkgs.bash}/bin/bash
+          set -e
+          set -o pipefail
+          # set -u # We are not ready for this yet
+          ${exportSessionVariables}
+        '';
       in
-      pkgs.runCommand name { inherit script; } ''
-        echo '#!${pkgs.bash}/bin/bash' > "$out"
-        echo 'set -e' >> "$out"
-        #echo 'set -u' >> "$out" # we are not ready for this yet
-        echo 'set -o pipefail' >> "$out"
-        echo -n "$script" >> "$out"
```

Comment:
`echo -n` didn't originate from me, I've just imported `writeBash*` from elsewhere. My preferred way to generate scripts is
```
cat > $out <<\EOF
...
EOF
```
or `writeText`.